### PR TITLE
MRG: update prefetch/gather output to be clearer

### LINKED ISSUE #2539: check gather/prefetch `--threshold-bp` behavior in a few cases
This came up in conversation with @bryshalm or @hehouts -

* what happens when `--threshold-bp` is non-zero but smaller than the scaled value? IMO it should find all matches with any overlap. make sure there's a test here :).
* should we do a better job of alerting people to the (high) default value of --threshold-bp when no matches are found? (yes, I think so)

### PR DESCRIPTION
**Pull request by ctb**
Fixes https://github.com/sourmash-bio/sourmash/issues/2539

This PR:
* add tests for very low threshold (1) on gather
* Adds output to be clear about --threshold-bp when no matches are found: `No matches found for --threshold-bp at ...`

### CODE REVIEW

**Comment by ctb** [comment]
Ready for review @sourmash-bio/devs - will need to update to latest, and also running afoul of https://github.com/sourmash-bio/sourmash/issues/2544 in the tests.

File: src/sourmash/commands.py [hunk 1]
```diff
@@ -797,7 +797,8 @@ def gather(args):
             if prefetch_csvout_fp:
                 prefetch_csvout_fp.flush()
 
-        notify(f"Found {len(save_prefetch)} signatures via prefetch; now doing gather.")
+        display_bp = format_bp(args.threshold_bp)
+        notify(f"Prefetch found {len(save_prefetch)} signatures with overlap >= {display_bp}.")
```
Comment (bluegenes) [inline src/sourmash/commands.py:801]: should we notify anything about https://github.com/sourmash-bio/sourmash/issues/2318?

Comment (ctb) [reply src/sourmash/commands.py:801]: I think linking this PR into that issue is enough? :) yeah that requires a whole different set of foo!

**Review by bluegenes — APPROVED** [review]
lgtm!

**Pull request merged**